### ISSUE TITLE
VsoClient is swallowing exceptions thrown in new Promises

### ISSUE THREAD

**Issue by stevek-axo** [Microsoft/vsts-node-api#142]
```javascript
const vsts = require('vso-node-api');

const accessToken = 'TOKEN';
const url = 'https://URL.visualstudio.com';

const accessTokenHandler = vsts.getPersonalAccessTokenHandler(accessToken);
const connection = new vsts.WebApi(url, accessTokenHandler)

connection.getIdentitiesApi()
  .then((identitiesApi) => {
    return identitiesApi.getSelf();
  })
  .then((self) => {
    console.log(self);
  })
  .catch((error) => {
    console.log(error);
  });
```

yields the following error:

![screen shot 2018-02-13 at 2 48 30 pm](https://user-images.githubusercontent.com/35500807/36175528-fe7bf712-10cc-11e8-9307-110d25424f6e.png)

Despite the above catch block, the error message states that the Promise rejection was unhandled. After doing some digging around in the stack trace, I traced the issue to the [`new Promise` construction in VsoClient.ts](https://github.com/Microsoft/vsts-node-api/blob/d467665e8ce692b5d9f10409a928a9e406c2428a/api/VsoClient.ts#L120). The error is swallowed inside the Promise and never bubbles up to the `catch` block in the above code sample.

**Comment by bryanmacfarlane** [comment]
Yeah, it seems like that should have .then and .error.

BTW, try npm install vso-task-lib@preview - that should alleviate the root of the issue.  See recent samples.  But I agree that function should handle errors / exceptions.

**Comment by stevek-axo** [comment]
Do you mean `vso-node-api@preview`? I'm on the `6.3.0-preview` version.

**Comment by bryanmacfarlane** [comment]
Yes.  Can you outline the repro and scenario where that is throwing?  A repro script?  Or was it via code inspection.  We will of course fix but I want to make sure we have a repro to validate.

**Comment by stevek-axo** [comment]
I think it should be reproducible with the code snippet in the original post. The personal access token I used had all scopes available to it, and the URL is my own workspace.

**Comment by stephenmichaelf** [comment]
Should be resolved now in version 6.5.3. Will publish soon.
